runtime/foucoco/src/lib.rs

```diff
@@ -893,6 +900,517 @@ parameter_types! {
+				);
+				error!("allowance_request : {:#?}", allowance_request);
+				error!("allowance : {:#?}", allowance);
+
+				env.write(&allowance.encode(), false, None)
+					.map_err(|_| DispatchError::Other("ChainExtension failed to call balance"))?;
+			},
+
+			//TODO perhaps we need this functionality. if not. will remove it.
+			//increase_allowance/decrease_allowance
+			// 1111 => {
+			// 	use frame_support::dispatch::DispatchResult;
+			//     let mut env = env.buf_in_buf_out();
+			//     let request: (u32, [u8; 32], [u8; 32], u128, bool) = env.read_as()?;
+			// 	let (asset_id, owner, delegate, amount, is_increase) = request;
+			// 	let mut vec = &owner.to_vec()[..];
+			// 	let owner_address = AccountId::decode(&mut vec).unwrap();
+			// 	let mut vec = &delegate.to_vec()[..];
+			// 	let delegate_address = AccountId::decode(&mut vec).unwrap();
+
+			// 	use crate::sp_api_hidden_includes_construct_runtime::hidden_include::traits::fungibles::approvals::Inspect;
+			//     let allowance :u128 = Assets::allowance(asset_id, &owner_address, &delegate_address);
+			// 	let new_allowance =
+			// 	if is_increase {allowance + amount}
+			// 	else {
+			// 		if allowance < amount  { 0 }
+			// 		else {allowance - amount}
+			// 	};
+			// 	let cancel_approval_result = pallet_assets::Pallet::<Runtime>::
+			// 	cancel_approval(Origin::signed(owner_address.clone()),
+			// 	asset_id,
+			// 	MultiAddress::Id(delegate_address.clone()));
+			// 	match cancel_approval_result {
+			// 		DispatchResult::Ok(_) => {
+			// 			error!("OK cancel_approval")
+			// 		}
+			// 		DispatchResult::Err(e) => {
+			// 			error!("ERROR cancel_approval");
+			// 			error!("{:#?}", e);
+			// 			let err = Result::<(),PalletAssetErr>::Err(PalletAssetErr::from(e));
+			// 			env.write(&err.encode(), false, None).map_err(|_| {
+			// 				DispatchError::Other("ChainExtension failed to call 'approve transfer'")
+			// 			})?;
+			// 		}
+			// 	}
+			// 	if cancel_approval_result.is_ok(){
+			// 		let approve_transfer_result = pallet_assets::Pallet::<Runtime>::
+			// 		approve_transfer(Origin::signed(owner_address),
+			// 		asset_id,
+			// 		MultiAddress::Id(delegate_address),
+			// 		new_allowance);
+			// 		error!("old allowance {}", allowance);
+			// 		error!("new allowance {}", new_allowance);
+			// 		error!("increase_allowance input {:#?}", request);
+			// 		error!("increase_allowance output {:#?}", approve_transfer_result);
+			// 		match approve_transfer_result {
+			// 			DispatchResult::Ok(_) => {
+			// 				error!("OK increase_allowance")
+			// 			}
+			// 			DispatchResult::Err(e) => {
+			// 				error!("ERROR increase_allowance");
+			// 				error!("{:#?}", e);
+			// 				let err = Result::<(),PalletAssetErr>::Err(PalletAssetErr::from(e));
+			// 				env.write(&err.encode(), false, None).map_err(|_| {
+			// 					DispatchError::Other("ChainExtension failed to call 'approve transfer'")
+			// 				})?;
+			// 			}
+			// 		}
+			// 	}
+			// }
```

Comment:
What about this block/comment, can we remove it?

Reply:
Done